Code/GraphMol/FileParsers/MaeMolSupplier.cpp

```diff
@@ -0,0 +1,145 @@
+}
+
+MaeMolSupplier::MaeMolSupplier(const std::string &fileName, bool sanitize, bool removeHs) {
+  auto *ifs = new std::ifstream(fileName.c_str(), std::ios_base::binary);
+  if (!ifs || !(*ifs) || ifs->bad()) {
+    std::ostringstream errout;
+    errout << "Bad input file " << fileName;
+    throw BadFileException(errout.str());
+  }
+  dp_inStream = (std::istream *)ifs;
+  df_owner = true;
+  df_sanitize = sanitize;
+  df_removeHs = removeHs;
+
+  d_reader.reset(new schrodinger::mae::Reader(*ifs));
+  d_next_struct = d_reader->next("f_m_ct");
+}
+
+void MaeMolSupplier::init() {}
+void MaeMolSupplier::reset() {}
+
+ROMol *MaeMolSupplier::next() {
+  if (d_next_struct == nullptr) {
+    throw FileParseException("All structures read from Maestro file");
+  }
+
+  auto mol = new RWMol();
+  auto mol_title = d_next_struct->getStringProperty("s_m_title");
+  mol->setProp(common_properties::_Name, mol_title);
+  std::map<int, Atom *> amap;
+  // Atom data is in the m_atom indexed block
+  {
+    const auto atom_data = d_next_struct->getIndexedBlock("m_atom");
+    // All atoms are gauranteed to have these three field names:
+    const auto atomic_numbers = atom_data->getIntProperty("i_m_atomic_number");
+    const auto xs = atom_data->getRealProperty("r_m_x_coord");
+    const auto ys = atom_data->getRealProperty("r_m_y_coord");
+    const auto zs = atom_data->getRealProperty("r_m_z_coord");
+    const auto size = atomic_numbers->size();
+    std::shared_ptr<schrodinger::mae::IndexedIntProperty> atomic_charges;
+    try {
+      atomic_charges = atom_data->getIntProperty("i_m_formal_charge");
+    } catch(std::out_of_range& e) { }
+
+    // atomic numbers, and x, y, and z coordinates
+    auto conf = new RDKit::Conformer(size);
+    conf->set3D(true);
+    conf->setId(0);
+    for (size_t i=0; i<size; ++i) {
+      Atom *atom = new Atom(atomic_numbers->at(i));
+      amap[i] = atom;
+      mol->addAtom(atom, true, true);
+      if(atomic_charges) {
+          atom->setFormalCharge(atomic_charges->at(i));
+      }
+
+      RDGeom::Point3D pos;
+      pos.x = xs->at(i);
+      pos.y = ys->at(i);
+      pos.z = zs->at(i);
+      conf->setAtomPos(i, pos);
+    }
+    mol->addConformer(conf, false);
+  }
+
+  // Bond data is in the m_bond indexed block
+  {
+    const auto bond_data = d_next_struct->getIndexedBlock("m_bond");
+    // All bonds are gauranteed to have these three field names:
+    auto from_atoms = bond_data->getIntProperty("i_m_from");
+    auto to_atoms = bond_data->getIntProperty("i_m_to");
+    auto orders = bond_data->getIntProperty("i_m_order");
+    const auto size = from_atoms->size();
+
+    for (size_t i=0; i<size; ++i) {
+      // Maestro atoms are 1 indexed!
+      const auto from_atom = from_atoms->at(i) - 1;
+      const auto to_atom = to_atoms->at(i) - 1;
+      const auto order = bolookup.find(orders->at(i))->second;
+      if(from_atom > to_atom) continue; // Maestro files double-list bonds
```

Comment:
We do support dative bonds...but I don't think in that way, @mbeachy what am I missing here in how we do this?  My understanding is that our doubling of bonds is because in ancient times we had the concept of a 'half bond', so if you only define one side in the format, you only get half of the bond drawn in Maestro.  Mike please correct me if I'm wrong about this.